File: api/app/Models/Plan.php
<?php
// app/Models/Plan.php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Plan extends Model
{
    protected $fillable = [
        'code', 'name', 'monthly_price', 'max_users', 'max_elevators',
        'sms_quota', 'features', 'is_active',
    ];

    protected function casts(): array
    {
        return [
            'monthly_price' => 'decimal:2',
            'features'      => 'array',
            'is_active'     => 'boolean',
        ];
    }
}

File: api-net/Models/Plan.cs
namespace Liftonom.Api.Models;

public class Plan
{
    public long Id { get; set; }
    public string Code { get; set; } = "";
    public string Name { get; set; } = "";
    public decimal MonthlyPrice { get; set; }
    public int? MaxUsers { get; set; }
    public int? MaxElevators { get; set; }
    public int? SmsQuota { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}
